src/app/vault/ciphers.component.html

```diff
@@ -6,11 +6,9 @@
     </div>
 </div>
 <div class="content">
-    <ng-container *ngIf="(isPaging() ? pagedCiphers : ciphers) as filteredCiphers">
-        <div class="list" *ngIf="filteredCiphers.length" infiniteScroll [infiniteScrollDistance]="1"
-            [infiniteScrollContainer]="'#items .content'" [fromRoot]="true" [infiniteScrollDisabled]="!isPaging()"
-            (scrolled)="loadMore()">
-            <a *ngFor="let c of filteredCiphers" appStopClick (click)="selectCipher(c)"
+    <cdk-virtual-scroll-viewport itemSize="42" minBufferPx="300" maxBufferPx="300" *ngIf="ciphers.length">
```

Comment:
shouldn't the buffer size be a multiple of the itemsize?

Also, in my desktop env the calculated item height is 43 before these changes, can we calculate this or is precision not really that important?

Reply:
The buffer size doesn't need to be a multiple of the itemSize, it will just render in multiples of the itemSize without exceeding the buffer size. That said, I didn't think through these buffer settings and I'm not sure they're doing anything, so I'll reevaluate here.

The calculated height is 43 for me as well - I had copied `42` from browser without checking. `ItemSize` is used to correctly size the `cdk-virtual-scroll-viewport`, so that the scrollbar matches the expected number of items. Any inaccuracy here could be multiplied over many items and result in the scrollbar being too long (leaving blank space at the end of the list) or too short (cutting off items). I'll look into grabbing the actual height of the first item instead as I don't feel great about hardcoded pixel values.

Reply:
Fixed by the custom scrolling strategy we've discussed